### PR TITLE
[BugFix] Fix fencing failed when a new leader is elected

### LINKED ISSUE #12136: [BUG] Fencing failed when a new leader is elected
### Steps to reproduce the behavior (Required)
Restart a 3 fe cluster continuously

### Expected behavior (Required)
Cluster runs normally

### Real behavior (Required)
Cluster crashes with error
```
2022-09-26 15:21:56,043 ERROR (stateListener|76) [Catalog.transferToMaster():1251] fencing failed. will exit.
```

### StarRocks version (Required)
 - all

### PR DESCRIPTION
**Pull request by gengjun-git**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #12136

## Problem Summary(Required) ：

Because of consistency bug in bdb(may arise when all node restart continuously), the data in epochDB may be discontinuous, so the `epochDB.count() + 1` can be an existing value, which will cause epochDb.putNoOverwrite failed. To fix bdb bug is difficulty, so we define the new epoch by `the largest value in epochDB + 1`.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 7 / 9 (77.78%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/ha/BDBHA.java|7|9|77.78%|[121, 122]|

File: fe/fe-core/src/main/java/com/starrocks/ha/BDBHA.java [hunk 1]
```diff
@@ -77,8 +82,7 @@ public boolean fencing() {
 
         for (int i = 0; i < RETRY_TIME; i++) {
             try {
-                long count = epochDb.getDb().count();
-                long myEpoch = count + 1;
+                long myEpoch = getLargestEpoch(epochDb.getDb()) + 1;
```
Comment (nshangyiming) [inline fe/fe-core/src/main/java/com/starrocks/ha/BDBHA.java:85]: Is it safe to do so?

File: fe/fe-core/src/main/java/com/starrocks/ha/BDBHA.java [hunk 2]
```diff
@@ -106,6 +110,19 @@ public boolean fencing() {
         return false;
     }
 
+    private long getLargestEpoch(Database epochDB) {
```
Comment (padmejin) [inline fe/fe-core/src/main/java/com/starrocks/ha/BDBHA.java:113]: I think `getLatestEpoch` is a better name. I suggest you add an asert to ensure that the latest epoch >= db.count().

File: fe/fe-core/src/main/java/com/starrocks/ha/BDBHA.java (hunk 1 above)
Comment (gengjun-git) [reply fe/fe-core/src/main/java/com/starrocks/ha/BDBHA.java:85]: Yes, this process is before all write ops.

**Review by padmejin — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 7 / 11 (63.64%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/ha/BDBHA.java|7|11|63.64%|[122, 123, 124, 125]|

**Review by nshangyiming — APPROVED** [review]

**Comment by nshangyiming** [comment]
run starrocks_admit_test

**Review by meegoo — APPROVED** [review]

**Review by Astralidea — APPROVED** [review]

**Comment by gengjun-git** [comment]
@Mergify backport-branch-2.2 backport-branch-2.3 backport-branch-2.4

**Pull request merged**